From 54a8632d899d7a4fa6376c5f658c0da99e9e0b78 Mon Sep 17 00:00:00 2001
From: Markus Armbruster <armbru@redhat.com>
Date: Fri, 17 Jan 2014 17:07:52 +0100
Subject: [PATCH 02/11] Revert "qdev: Do not let the user try to device_add when it cannot work"

RH-Author: Markus Armbruster <armbru@redhat.com>
Message-id: <1389978479-30595-3-git-send-email-armbru@redhat.com>
Patchwork-id: 56792
O-Subject: [PATCH 7.0 qemu-kvm 2/9] Revert "qdev: Do not let the user try to device_add when it cannot work"
Bugzilla: 669524
RH-Acked-by: Eduardo Habkost <ehabkost@redhat.com>
RH-Acked-by: Michael S. Tsirkin <mst@redhat.com>
RH-Acked-by: Marcel Apfelbaum <marcel.a@redhat.com>

This reverts commit de651299014ae67b110a2abbff4a9e544b215b2b.

This one is actually fine, but its non-trivial conflict resolution
would cause many more conflicts in the next few patches.  I'm going to
reapply the upstream commit in a place where it applies cleanly.

Signed-off-by: Markus Armbruster <armbru@redhat.com>
---
 qdev-monitor.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 qdev-monitor.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/qdev-monitor.c b/qdev-monitor.c
index f78ff64..d187db5 100644
--- a/qdev-monitor.c
+++ b/qdev-monitor.c
@@ -477,13 +477,13 @@ DeviceState *qdev_device_add(QemuOpts *opts)
         }
     }
 
-    k = DEVICE_CLASS(obj);
-
-    if (!k || k->cannot_instantiate_with_device_add_yet) {
+    if (!obj) {
         qerror_report(QERR_INVALID_PARAMETER_VALUE, "driver", "device type");
         return NULL;
     }
 
+    k = DEVICE_CLASS(obj);
+
     /* find bus */
     path = qemu_opt_get(opts, "bus");
     if (path != NULL) {
-- 
1.7.1

